zap/src/main/java/org/parosproxy/paros/core/scanner/AppParametersBuilder.java

```diff
@@ -0,0 +1,37 @@
+package org.parosproxy.paros.core.scanner;
+
+import java.util.ArrayList;
+import java.util.Collections;
+import java.util.HashSet;
+import java.util.List;
+import java.util.Set;
+import org.apache.logging.log4j.LogManager;
+import org.apache.logging.log4j.Logger;
+
+/**
+ * {@code AppParametersBuilder} provides an easy way to build {@code AppParameters} and validates
+ * the provided inputs.
+ *
+ * @author preetkaran20@gmail.com KSASAN
+ */
+public class AppParametersBuilder {
+
+    private List<AppParameter> appParameters = new ArrayList<>();
+    private Set<Integer> nameValuePairPositions = new HashSet<>();
+    private final Logger logger = LogManager.getLogger(this.getClass());
+
+    public AppParametersBuilder addAppParameter(
```

Comment:
It should be possible to set the name/value separately.

Reply:
Didn't get you, Can you please explain more?

Reply:
The name can be an input vector as well and should be possible to change it without changing the value.

Reply:
ok, so you are suggesting the addition of one more method with just name. Right?

Reply:
Yeah, one method for the value and another for the name (I guess we could still provide this one for the cases both name/value are changed).